test/unittest/StatisticsBackend/GetDataTests.cpp

```diff
@@ -0,0 +1,4340 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <cmath>
+
+#include <exception/Exception.hpp>
+#include <Monitor.hpp>
+#include <StatisticsBackend.hpp>
+#include <StatisticsBackendData.hpp>
+#include <database/database.hpp>
+#include <types/types.hpp>
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::AnyNumber;
+using ::testing::Throw;
+
+using namespace eprosima::statistics_backend;
+using namespace eprosima::statistics_backend::database;
+
+
+
+class get_data_tests_base
+{
+
+public:
+
+    get_data_tests_base()
+    {
+        details::StatisticsBackendData::get_instance()->database_.reset(new Database);
+        StatisticsBackend::load_database("resources/database_dump.json");
+    }
```

Comment:
It would have been clearer to reset the database in a `TearDown()` method and load the database in a `SetUp()` one

Reply:
Future improvement